libtenzir/include/tenzir/view3.hpp

```diff
@@ -215,6 +218,17 @@ inline auto record_view3::iterator::operator*() const
   };
 }
 
+inline auto record_view3::field(std::string_view name)
+  -> std::optional<data_view3> {
+  for (auto [index, field] :
+       detail::enumerate<int>(array_.struct_type()->fields())) {
+    if (field->name() == name) {
+      return view_at(*array_.field(index), index_);
+    }
+  }
+  return {};
+}
```

Comment:
This really invites bad perf because it does the name lookup for every single event. Is there a way that we can avoid this and do the name lookup only once per batch? If we can get by with not having this function than I think it shouldn't exist, or it should be called `field_with_bad_performance` or similar.

Reply:
> Is there a way that we can avoid this and do the name lookup only once per batch?

I ended up not using this function anyway, exactly because of the lookup perf hit. I think it can still be useful, and `data`/`data_view` have similar operations after all. But I'll just remove it, we can reintroduce it when needed.